### PR TITLE
Added docker development environment.

### PR DESCRIPTION
**Pull request by martinespinach**
* Technical improvement.
* Details:
  - Docker image & container definition for development run.
  - Remote debug for tests.

- - - -

These changes:

* Add a new feature

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,9 @@ dist:
 test:
 	openfisca-run-test --country-package openfisca_barcelona openfisca_barcelona/tests
 
+test_debug:
```
Comment (jvalduvieco) [inline Makefile]: `test_remote_debug` so it is more explicit?

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+FROM python:2.7-alpine
+
+RUN apk update && \
+	apk add build-base && \
+	apk add linux-headers && \
+	apk add yaml-dev && \
```
Comment (jvalduvieco) [inline docker/Dockerfile:8]: Why not use `setup.py`? `pip install -e ".[test]"`

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	pip install nose && \
+	pip install pydevd
+
+EXPOSE 2000
+
+COPY ./docker-entrypoint.sh /
+
+WORKDIR /usr/src/app
+
+ENTRYPOINT [ "/docker-entrypoint.sh" ]
```
Comment (jvalduvieco) [inline docker/Dockerfile]: Add newline at EOF plz.
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: docker/docker-entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+	pip install -e .
+fi
+
+grep "^PATHS_FROM_ECLIPSE_TO_PYTHON =" /usr/local/lib/python2.7/site-packages/pydevd_file_utils.py
+if [ $? -ne 0 ]; then
+	PARENT_PATH=${HOST_PATH%/*}
+	sed -i "s@setup_client_server_paths(PATHS_FROM_ECLIPSE_TO_PYTHON)@PATHS_FROM_ECLIPSE_TO_PYTHON = [\(r'$PARENT_PATH',r'/usr/src/app'\)]\nsetup_client_server_paths(PATHS_FROM_ECLIPSE_TO_PYTHON)@" /usr/local/lib/python2.7/site-packages/pydevd_file_utils.py
+fi
+
+exec openfisca serve --country-package $COUNTRY_PACKAGE --bind 0.0.0.0:2000
```
Comment (jvalduvieco) [inline docker/docker-entrypoint.sh]: Add NL at EOF plz

File: docker/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+## Development environment using Docker
```
Comment (jvalduvieco) [inline docker/README.md]: Should we add this info at root `README.md` so it's easier to find? If not should we link from there?

**Review by jvalduvieco — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (martinespinach) [reply Makefile]: ok

File: docker/Dockerfile (hunk 2 above)
Comment (martinespinach) [reply docker/Dockerfile:8]: Apk packages are alpine packages needed inside container to allow python extensions installation. But if you are referring to the pip install packages that are below, yes, it is possible to use the setup.py. I had initially implemented this Dockerfile outside of the project, but being inside it makes more sense to use the setup.py.

File: docker/Dockerfile (hunk 3 above)
Comment (martinespinach) [reply docker/Dockerfile]: Ok. I am more in favor of: "Today, the newline is not required anymore. Sure, many apps still have problems if the newline is not there, but I'd consider that a bug in those apps." But there's no problem.

File: docker/docker-entrypoint.sh (hunk 4 above)
Comment (martinespinach) [reply docker/docker-entrypoint.sh]: Ok.

File: docker/README.md (hunk 5 above)
Comment (martinespinach) [reply docker/README.md]: Yes of course. I had not added it because I wanted to keep it separate for the review.

File: docker/Dockerfile (hunk 3 above)
Comment (jvalduvieco) [reply docker/Dockerfile]: Make a request to change the POSIX standard... ;)

**Review by jvalduvieco — APPROVED** [review]

**Pull request merged**